bots/csr/src/main/java/org/openjdk/skara/bots/csr/PullRequestWorkItem.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.skara.bots.csr;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+import org.openjdk.skara.bot.WorkItem;
+import org.openjdk.skara.forge.HostedRepository;
+import org.openjdk.skara.forge.PullRequest;
+import org.openjdk.skara.issuetracker.Issue;
+import org.openjdk.skara.issuetracker.IssueProject;
+import org.openjdk.skara.jbs.Backports;
+import org.openjdk.skara.jbs.JdkVersion;
+import org.openjdk.skara.jcheck.JCheckConfiguration;
+
+/**
+ * The PullRequestWorkItem is the work horse of the CSRBot. It gets triggered when
+ * either the pull request itself, or any CSR issue associated with it have been
+ * updated. It operates on one single pull request and re-evaluates the CSR state
+ * for it.
+ */
+class PullRequestWorkItem implements WorkItem {
+    private final static String CSR_LABEL = "csr";
+    private final static String CSR_UPDATE_MARKER = "<!-- csr: 'update' -->";
+    private static final String PROGRESS_MARKER = "<!-- Anything below this marker will be automatically updated, please do not edit manually! -->";
```

Comment:
It's a bit unfortunate to duplicate string literals, especially those used as markers. Can you import this from `CheckRun.progressMarker` instead?

Reply:
I completely agree that duplicated strings are bad. Importing directly from another bot module wouldn't be great IMO. I will look into moving this into some common class in a shared module instead. Note though that this isn't new in this patch.

Reply:
Actually, thinking on this some more, can I file a followup to attack constants shared between bots in a more uniform way? I think many of these should be shared knowledge between bots, including label names, so adding a PullRequestContants class in the forge module (next to PullRequestUtils) would be a good idea I think. I don't want to pollute this change with this though.

Reply:
Filed https://bugs.openjdk.org/browse/SKARA-1547

Reply:
I saw that this was not new. Agree that it's better to fix in a separate PR.